server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -271,8 +302,11 @@ public SourceFieldMapper build() {
                     );
                 }
             }
+
+            final Mode settingsSourceMode = INDEX_MAPPER_SOURCE_MODE_SETTING.get(settings);
```

Comment:
Here I need to know if the `index.mapper.source.mode` setting was defined or not to be able to decide if the `_source.mode` can be used. Of course after deprecating it this logic will change.